### DISCUSSION TITLE
libp2p over DMR radio

### DISCUSSION THREAD

**Discussion by imsergiobernal** [libp2p/specs#641] [Ideas]
Hello, I did not found anything about DMR implementation of libp2p. I think that this implementation would level up the P2P resiliance and adaptation. I'm in a flood and denied zone, and all internet went down. I'm new over here, just curious.

**Comment by hsanjuan** [comment]
Hola. I hope all is well with your close ones.

I have not heard of such thing. libp2p is "transport agnostic" in principle, so one could technically build this by making a new transport that complies with the Transport interface: https://github.com/libp2p/go-libp2p/blob/master/core/transport/transport.go. Probably with a few caveats but I'm not familiar with DMR to tell. It would be however a very interesting project.

**Comment by master255** [comment]
You can build a bridge to the internet by running [DHT Money](https://t.me/dhtmoney/3) apps. And have internet using a proxy server (built into DHT Money) that is located somewhere where there is internet.

So all you need is one Android phone with internet and you will have internet for everyone.

And with DHT Money, which runs on Libp2p - the internet will be transferred between all phones where DHT Money is installed, without any settings. It's enough just to install the application.

Access to the built-in proxy is by ID. And this, of course, it's necessary to configure.
